model/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/model/value/primitive/DateTimeValue.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 Fraunhofer IOSB, eine rechtlich nicht selbstaendige
+ * Einrichtung der Fraunhofer-Gesellschaft zur Foerderung der angewandten
+ * Forschung e.V.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.fraunhofer.iosb.ilt.faaast.service.model.value.primitive;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class DateTimeValue extends TypedValue<ZonedDateTime> {
+
+    public static final String DEFAULT_TIMEZONE = "UTC";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DateTimeValue.class);
+
+    public DateTimeValue() {
+        super();
+    }
+
+
+    public DateTimeValue(ZonedDateTime value) {
+        super(value);
+    }
+
+
+    @Override
+    public String asString() {
+        return value != null ? value.toString() : "";
+    }
+
+
+    @Override
+    public void fromString(String value) throws ValueFormatException {
+        if (value == null) {
```

Comment:
null check can be moved to super class

Reply:
How exactly should that be realized? 
If I move the whole "if null check" to the super class, I don't know in the derived class if the super class handled the setValue?

Reply:
Move `if (value == null) {...}` to superclass and then call super.fromString() here instead. As the inheritance hierarchy is fixed we know for sure that superclass only checks for null and can handle actual parse in the concrete classes.

Reply:
I'm not sure if I gotz that right. 
I moved the whole if block to the super class. The super class looks like that now: 
`        if (value == null) {
            this.setValue(null);
        }
`
The derived classes like DateTimeValue look like that; 
`        super.fromString(value);
        try {
            this.setValue(ZonedDateTime.parse(value));
        }
...
`
But that doesn't work, because the setValue in the derived class is also called, when value == null. 
Or do we have a misunderstanding here?

Reply:
You are correct. Only option I see is to add `if (value != nul) {...}` in the subclasses

Reply:
OK,that works. 
But in my opinion that doesn't make sense. 
Instead of directly call setValue in the derived class, we call the super class, where we have again the same check `if (value == null)`. 
I think it would be better and clearer if we leave fromString in the super class abstract, and directly call setValue in the derived class, as it is in the current commit. 
Sonar is also satisfied with this version.

Reply:
That's also fine with me